### PR TITLE
[BIOMAGE-1793] - Create RDS environments for staging

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Problem
We want to have a separate database to test changes for our own environment.

#### Solution
Stage a new RDS resource for staging environment suffixed with a sandbox id.

#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1793

#### Link to staging deployment URL 
In API PR

#### Links to any Pull Requests related to this
- IAC - https://github.com/hms-dbmi-cellenics/iac/pull/318
- API - https://github.com/hms-dbmi-cellenics/api/pull/323
- Biomage utils - https://github.com/hms-dbmi-cellenics/biomage-utils/pull/81

#### Anything else the reviewers should know about the changes here
- Deploy a new RDS resource for every staging environment.

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock  (with latest production data downloaded with biomage experiment pull)
- [X] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/hms-dbmi-cellenics/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/hms-dbmi-cellenics/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/hms-dbmi-cellenics/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: cf/rds.yaml [hunk 1]
```diff
@@ -57,8 +57,8 @@ Resources:
   DBInstance1:
     Type: AWS::RDS::DBInstance
     Properties:
-      DBInstanceClass: db.t3.medium
+      DBInstanceClass: db.t4g.medium
```
Comment (kafkasl) [inline cf/rds.yaml:60]: why this change?

File: cf/test-rds.yaml [hunk 2]
```diff
@@ -14,6 +14,10 @@ Parameters:
     Type: String
     Default: default
     Description: The sandbox ID of the environment that this cluster is created for.
+  DBInstanceType:
+    Type: String
+    Default: db.t4g.medium
```
Comment (kafkasl) [inline cf/test-rds.yaml:19]: use the smallest one for testing, we discussed that for the staging envs we should use smaller instances (not only now during development)

Comment (aerlaut) [reply cf/test-rds.yaml:19]: db.t4g.medium is the smallest instance that we can use with Postgres 13.4. Like the t3, it is only [recommended to use this instance for non-production uses](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.DBInstanceClass.html). However, It is [a little bit cheaper](https://aws.amazon.com/rds/postgresql/pricing/) than t3. That's why I use this instance for staging.

As pictured in this table ([ref](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.DBInstanceClass.html)), the `.medium` instances are the smallest instance that we can use for the postgres version that we use (13.4). The left column is Aurora MySQL, the right column is for Aurora PostgreSQL.

image

File: cf/rds.yaml (hunk 1 above)
Comment (aerlaut) [reply cf/rds.yaml:60]: db.t4g.medium is the smallest instance that we can use with Postgres 13.4. It is like the t3, but [a little bit cheaper](https://aws.amazon.com/rds/postgresql/pricing/). Actually this instance is not recommended for [production use](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.DBInstanceClass.html).

The definition in this file governs the instance for both production and staging. Maybe we should use a bigger instance for production?

File: cf/test-rds.yaml (hunk 2 above)
Comment (kafkasl) [reply cf/test-rds.yaml:19]: how inconvenient we didn't notice that when we choose postgre over mysql. could you check how much will it cost to run 1 instance for a day?

**Review by kafkasl — APPROVED** [review]
